stan/math/prim/meta/forward_as.hpp

```diff
@@ -31,7 +31,8 @@ constexpr bool eigen_static_size_match(T1 desired, T2 actual) {
  */
 template <typename T_desired, typename T_actual,
           typename
-          = std::enable_if_t<std::is_convertible<T_actual, T_desired>::value
+          = std::enable_if_t<std::is_same<std::decay_t<T_actual>,
```

Comment:
To make this work I had to change the is_convertible logic to is_same:

```
double x = 1.0;
forward_as<var>(x).adj() += 2.0;
```

double can convert to var, so in the old logic `decltype(forward_as<var>(1.0))` is a double. Also I didn't see tests for `forward_as`. There should be tests, but I can add those after we decide what `forward_as` should do for sure.